lib/order.rb

```diff
@@ -0,0 +1,43 @@
+class Order
+  
+  attr_reader :total
+  attr_accessor :status
+  
+  def initialize
+    @status = Array.new
+    @total = 0
+  end
+  
+  def add(item)
+    status << item
+    calculate_partial
+  end
+  
+  def check_order
+    calculate_basket
+    calculate_total
+  end
+  
+  def clear_order
+    self.status = Array.new
+    "Your basket is now empty."
+  end
+  
+  private
+  
+  def calculate_basket
+    basket = status.group_by { |item| item.keys }
```

Comment:
Style/SymbolProc: Pass `&:keys` as an argument to `group_by` instead of a block.